src/main/java/de/retest/recheck/RecheckAdapter.java

```diff
@@ -12,6 +12,15 @@
  */
 public interface RecheckAdapter {
 
+	/**
+	 * Returns an instance of the RecheckAdapter, that has been initialized with the given {@link RecheckOptions}. In
+	 * case more specific recheck options are used during initialization of the {@link RecheckImpl}, then these are
+	 * passed on (and need to be cast).
+	 */
+	default RecheckAdapter initialize( final RecheckOptions opts ) {
```

Comment:
So, this is essentially a factory method (as you mentioned, we should have a proper factory class for this). 

I would rename this method and javadoc to underline, that we expect a **new** RecheckAdapter (as setting fields is not the ideal way). I would even suggest to have this incorporated into `convert` as parameter (via another default method)?

```
default Set<RootElement> convert(Object toVerify, RecheckOptions options) {
    return check(toVerify);
}
```

Secondly, I am not sure, if we should expose all options to the adapter, as this may confuse people.

Reply:
No, we do not want this in the convert method.
How would you rename the method?

Reply:
> No, we do not want this in the convert method.

Now we create (ideally) a new instance of a `RecheckAdapter` for each `check`&mdash;though the options do not change with the lifetime of the `Recheck` instance&mdash;,which seems to be a bit excessive. Why not just pass this along with the check and have only one single adapter instance?

> How would you rename the method?

Well, now the initial `RecheckAdapter` which is created with the default constructor through the `ServiceLoader` only acts as a factory (as described above). Thus keep to some default like `create`.

Reply:
Hmm. I was just renaming this method when I realized that the current approach is better. We currently do not enforce creating a new instance. So if for some reason an adapter is heavyweight (which currently is not the case), it is up to the adapter to decide whether he wants to return a fresh instance here, or just change some internal values. I thus would keep the old method name. That also addresses your concern about the lifecycle.

Reply:
Could you comment on your aversion for an additional parameter. This would not break the API, save on objects being created and remove the need for a factory.

> So if for some reason an adapter is heavyweight (which currently is not the case)

We just do some IO and read the default.yml and attribtute.yml every time the adapter is created. I would not call this lightweight.